src/common/actions/repository-input.js

```diff
@@ -52,3 +56,49 @@ export function verifyGitHubRepositoryError(error) {
     error: true
   };
 }
+
+export function createSnap(repository, location) {
+  return (dispatch) => {
+    if (repository) {
+      dispatch({
+        type: CREATE_SNAP,
+        payload: repository
+      });
+
+      const BASE_URL = conf.get('BASE_URL');
```

Comment:
I've been getting out of the habit of doing that for configuration items, because it makes testing harder in cases where we need to vary the configuration; but it's true that in this case that doesn't matter.  I've moved this one.